tutorials/tutorial1/tutorial.py

```diff
@@ -28,106 +28,116 @@
-# * `SpatialProblem` $\rightarrow$ spatial variable(s) presented in the differential equation
-# * `TimeDependentProblem` $\rightarrow$ time variable(s) presented in the differential equation
-# * `ParametricProblem` $\rightarrow$ parameter(s) presented in the differential equation
+# In summary, using PINA, we can initialize a problem with a class which inherits from three base classes: `SpatialProblem`, `TimeDependentProblem`, `ParametricProblem`, depending on the type of problem we are considering. For reference:
+# * `SpatialProblem` $\rightarrow$ a differential equation with spatial variable(s)
+# * `TimeDependentProblem` $\rightarrow$ a time-dependent differential equation
+# * `ParametricProblem` $\rightarrow$ a parametrized differential equation
+
 # 
 
-# ### Write the problem class
+# ### Write the `Problem` class
 # 
-# Once the problem class is initialized we need to write the differential equation in PINA language. For doing this we need to load the pina operators found in `pina.operators` module. Let's again consider the Equation (1) and try to write the PINA model class:
+# Once the `Problem` class is initialized, we need to represent the differential equation in PINA. In order to do this, we need to load the PINA operators from `pina.operators` module. Again, we'll consider Equation (1) and represent it in PINA:
 
 # In[14]:
 
 
 from pina.problem import SpatialProblem
 from pina.operators import grad
-from pina import Condition, Span
+from pina import Condition, CartesianDomain
 
 import torch
 
 
 class SimpleODE(SpatialProblem):
 
     output_variables = ['u']
-    spatial_domain = Span({'x': [0, 1]})
+    spatial_domain = CartesianDomain({'x': [0, 1]})
 
     # defining the ode equation
     def ode_equation(input_, output_):
 
         # computing the derivative
         u_x = grad(output_, input_, components=['u'], d=['x'])
 
-        # extracting u input variable
+        # extracting the u input variable
         u = output_.extract(['u'])
 
-        # calculate residual and return it
+        # calculate the residual and return it
         return u_x - u
 
-    # defining initial condition
+    # defining the initial condition
     def initial_condition(input_, output_):
         
-        # setting initial value
+        # setting the initial value
         value = 1.0
 
-        # extracting u input variable
+        # extracting the u input variable
         u = output_.extract(['u'])
 
-        # calculate residual and return it
+        # calculate the residual and return it
         return u - value
 
-    # Conditions to hold
+    # conditions to hold
     conditions = {
-        'x0': Condition(Span({'x': 0.}), initial_condition),
-        'D': Condition(Span({'x': [0, 1]}), ode_equation),
+        'x0': Condition(CartesianDomain({'x': 0.}), initial_condition),
```

Comment:
does it run? You should pass the keywords arguments `location` and `equation`. Also initial contition is not a PINA equation!